main/src/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -637,10 +639,19 @@ static public InputStream tryOpenAsArchive(File file, String mimeType, String co
             } else if (fileName.endsWith(".kmz")) {
                 return new ZipInputStream(new FileInputStream(file));
             }
-        } catch (IOException e) {
+        } catch (IOException ignored) {
         }
         return null;
     }
+
+    private static boolean tarIsGZipped(File file) {
+        int magic = 0;
+        try (RandomAccessFile raf = new RandomAccessFile(file, "r");) {
+            magic = raf.read() & 0xff | ((raf.read() << 8) & 0xff00);
+        } catch (IOException ignored) {
+        }
+        return magic == GZIPInputStream.GZIP_MAGIC;
```

Comment:
This seems to be checking whether the file is gzipped, but that does not guarantee that the gzipped file is a tar archive, right? What happens when we run this with a simply gzipped file (without enclosing tar archive)?

Reply:
Let me try that out :)

Reply:
Very nice, thanks a lot! Perhaps we can just add one or two tests and it should be good to go?

Reply:
Yes! I am not sure how I will but I will :)
Probably get an idea from looking at the current tests.